### PR TITLE
Add React Native to CI examples

### PR DESCRIPTION
**Pull request by aryaemami59**
This PR:

  - [X] Adds React Native example to CI workflow.

### CODE REVIEW

File: .github/workflows/build-and-test-types.yml [hunk 1]
```diff
@@ -179,13 +189,20 @@ jobs:
...
+          distribution: 'temurin'
+
       - name: Check MSW version
         working-directory: ./redux-toolkit/examples/publish-ci/${{ matrix.example }}
         run: yarn why msw
 
       - name: Build example
         working-directory: ./redux-toolkit/examples/publish-ci/${{ matrix.example }}
-        run: yarn build
+        run: NODE_OPTIONS=--openssl-legacy-provider yarn build
```
Comment (EskiMojo14) [inline .github/workflows/build-and-test-types.yml:205]: huh, how come this is needed?

Comment (aryaemami59) [reply .github/workflows/build-and-test-types.yml:205]: Not sure tbh, but I remember we had to add it to RTK as well to get the CI to work.

Comment (EskiMojo14) [reply .github/workflows/build-and-test-types.yml:205]: hmm, okay

**Comment by EskiMojo14** [comment]
thanks!

**Pull request merged**